sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -228,4 +241,37 @@ protected String visitSQLFunction(String funcName, String[] inputs) {
   protected String visitUnexpectedExpr(Expression expr) throws IllegalArgumentException {
     throw new IllegalArgumentException("Unexpected V2 expression: " + expr);
   }
+
+  protected String visitOverlay(String funcName, String[] inputs) {
+    if (inputs.length <= 3) {
+      return funcName + "(" + inputs[0] + " PLACING " + inputs[1] + " FROM " + inputs[2] +")";
+    } else {
+      return funcName + "(" + inputs[0] + " PLACING " + inputs[1] + " FROM " + inputs[2] +
+              " FOR " + inputs[3] +")";
+    }
+  }
+
+  protected String visitTrim(String[] inputs) {
```

Comment:
We can add parameter `direction`

Reply:
If we add parameter direction, we need to create new class.